libcst/metadata/scope_provider.py

```diff
@@ -136,6 +227,34 @@ def record_global_overwrite(self, name: str) -> None:
     def record_nonlocal_overwrite(self, name: str) -> None:
         ...
 
+    def get_qualified_names_for(self, node: cst.CSTNode) -> Collection[QualifiedName]:
+        results = set()
+        full_name = _QualifiedNameUtil.get_full_name_for(node)
+        if full_name is None:
+            return results
+        parts = full_name.split(".")
+        for assignment in self[parts[0]]:
+            if isinstance(assignment, Assignment):
+                assignment_node = assignment.node
+
+                if isinstance(assignment_node, cst.Import) or isinstance(
+                    assignment_node, cst.ImportFrom
+                ):
+                    _QualifiedNameUtil.find_qualified_name_for_import_alike(
+                        assignment_node, parts, results
+                    )
+                else:
+                    _QualifiedNameUtil.find_qualified_name_for_non_import(
+                        assignment, parts, results
+                    )
+            elif isinstance(assignment, BuiltinAssignment):
+                results.add(
+                    QualifiedName(
+                        f"builtins.{assignment.name}", QualifiedNameSource.IMPORT
```

Comment:
It seems really iffy to say that this was sourced from an `IMPORT`. I think we should have a `QualifiedNameSource.BUILTIN`.

Reply:
Added QualifiedNameSource.BUILTIN.